src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenExpression.vb

```diff
@@ -4687,6 +4687,2832 @@ done
+            Dim cv = CompileAndVerify(compilation,
+                            expectedOutput:=<![CDATA[
+done
+]]>)
+            cv.VerifyIL("C1.DecimalToSByte", <![CDATA[
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.1
+  IL_0002:  call       "Function System.Math.Round(Decimal, System.MidpointRounding) As Decimal"
+  IL_0007:  call       "Function Decimal.op_Explicit(Decimal) As SByte"
+  IL_000c:  ret
+}
+]]>)
+            cv.VerifyIL("C1.DecimalToByte", <![CDATA[
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.1
+  IL_0002:  call       "Function System.Math.Round(Decimal, System.MidpointRounding) As Decimal"
+  IL_0007:  call       "Function Decimal.op_Explicit(Decimal) As Byte"
+  IL_000c:  ret
+}
+]]>)
+            cv.VerifyIL("C1.DecimalToShort", <![CDATA[
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.1
+  IL_0002:  call       "Function System.Math.Round(Decimal, System.MidpointRounding) As Decimal"
+  IL_0007:  call       "Function Decimal.op_Explicit(Decimal) As Short"
+  IL_000c:  ret
+}
+]]>)
+            cv.VerifyIL("C1.DecimalToUShort", <![CDATA[
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.1
+  IL_0002:  call       "Function System.Math.Round(Decimal, System.MidpointRounding) As Decimal"
+  IL_0007:  call       "Function Decimal.op_Explicit(Decimal) As UShort"
+  IL_000c:  ret
+}
+]]>)
+            cv.VerifyIL("C1.DecimalToInteger", <![CDATA[
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.1
+  IL_0002:  call       "Function System.Math.Round(Decimal, System.MidpointRounding) As Decimal"
+  IL_0007:  call       "Function Decimal.op_Explicit(Decimal) As Integer"
+  IL_000c:  ret
+}
+]]>)
+            cv.VerifyIL("C1.DecimalToUInteger", <![CDATA[
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.1
+  IL_0002:  call       "Function System.Math.Round(Decimal, System.MidpointRounding) As Decimal"
+  IL_0007:  call       "Function Decimal.op_Explicit(Decimal) As UInteger"
+  IL_000c:  ret
+}
+]]>)
+            cv.VerifyIL("C1.DecimalToLong", <![CDATA[
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.1
+  IL_0002:  call       "Function System.Math.Round(Decimal, System.MidpointRounding) As Decimal"
+  IL_0007:  call       "Function Decimal.op_Explicit(Decimal) As Long"
+  IL_000c:  ret
+}
+]]>)
+            cv.VerifyIL("C1.DecimalToULong", <![CDATA[
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.1
+  IL_0002:  call       "Function System.Math.Round(Decimal, System.MidpointRounding) As Decimal"
+  IL_0007:  call       "Function Decimal.op_Explicit(Decimal) As ULong"
+  IL_000c:  ret
+}
 ]]>)
         End Sub
```

Comment:
> End Sub [](http://example.com/codeflow?start=7&length=8)

Since the optimization relies on presence of some VB runtime members, it will be good to add tests for the affected scenarios when an embedded VB runtime is used by the compilation. See the following compilation option
```
        ''' <summary>
        ''' Creates a new VisualBasicCompilationOptions instance with a different Embed VB Core Runtime specified.
        ''' </summary>
        ''' <param name="value">The Embed VB Core Runtime setting. </param>        
        ''' <returns>A new instance of VisualBasicCompilationOptions, if the embed vb core runtime is different; otherwise current instance.</returns>        
        Public Shadows Function WithEmbedVbCoreRuntime(value As Boolean) As VisualBasicCompilationOptions
```